app/src/main/java/com/x8bit/bitwarden/data/platform/repository/SettingsRepositoryImpl.kt

```diff
@@ -337,6 +340,60 @@ class SettingsRepositoryImpl(
                     ?: DEFAULT_IS_SCREEN_CAPTURE_ALLOWED,
             )
 
+    override val allSettingsBadgeCountFlow: StateFlow<Int>
+        get() = combine(
+            allSecuritySettingsBadgeCountFlow,
+            allAutofillSettingsBadgeCountFlow,
+            transform = ::sumSettingsBadgeCount,
+        )
+            .stateIn(
+                scope = unconfinedScope,
+                started = SharingStarted.Lazily,
+                initialValue = 0,
+            )
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    override val allSecuritySettingsBadgeCountFlow: StateFlow<Int>
+        get() = authDiskSource
+            .activeUserIdChangesFlow
+            .filterNotNull()
+            .flatMapLatest {
+                // can be expanded to support multiple security settings
+                getShowUnlockBadgeFlow(userId = it)
+                    .map { showUnlockBadge ->
+                        listOf(showUnlockBadge)
+                    }
+                    .map { list ->
+                        list.count { badgeOnValue -> badgeOnValue }
+                    }
+            }
+            .stateIn(
+                scope = unconfinedScope,
+                started = SharingStarted.Lazily,
+                initialValue = 0,
+            )
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    override val allAutofillSettingsBadgeCountFlow: StateFlow<Int>
+        get() = authDiskSource
+            .activeUserIdChangesFlow
+            .filterNotNull()
+            .flatMapLatest {
+                // Can be expanded to support multiple autofill settings
+                getShowAutofillBadgeFlow(userId = it)
+                    .map { showAutofillBadge ->
+                        listOf(showAutofillBadge)
```

Comment:
Are these built with the idea that there could be more autofill badges in the future?

Reply:
yes, thinking about it like nesting, this allows for a badge per sub-setting in the autofill menu, like may we show one for the accessibility stuff like that. where in the future it could be:
```kotlin
combine(
    getShowAutofillBadgeFlow(),
    getShowAccessibilityBadgeFlow(),
) { a, b ->
        listOf(a,b)
}
.soOnSoForth()

Reply:
Makes sense! 👍